src/lib/efhw/af_xdp.c

```diff
@@ -1416,13 +1416,59 @@ af_xdp_nic_buffer_table_clear(struct efhw_nic *nic,
  * Filtering
  *
  *--------------------------------------------------------------------*/
+
+static int
+af_xdp_ethtool_set_rxfh_context(struct efhw_nic *nic, const u32 *indir,
+                                const u8 *key, u8 hfunc, u32 *rss_context,
+                                u8 delete)
+{
+  struct net_device *dev = nic->net_dev;
+  const struct ethtool_ops *ops = dev->ethtool_ops;
+
+  EFHW_ASSERT(rss_context);
+
+#ifndef EFRM_HAVE_SET_RXFH_CONTEXT
+  /* linux >= 6.8 removes ethtool_ops::set_rxfh_context(). We use set_rxfh(). */
+
+  int rc;
+  struct ethtool_rxfh_param rxfh = {
+    .hfunc = hfunc,
+    .indir_size = nic->rss_indir_size,
+    .key_size = nic->rss_key_size,
+    .rss_context = *rss_context,
+    .rss_delete = delete,
+  };
+
+  memcpy(rxfh.indir, indir, sizeof(indir[0]) * nic->rss_indir_size);
+  memcpy(rxfh.key, key, nic->rss_key_size);
```

Comment:
Wouldn't `rxfh.indir` and `rxfh.key` both be 0 here? So this would end up writing to null.

I'd think it would be fine just setting `.indir = indir` and `.key = key` in the initialiser above.

Reply:
> Wouldn't `rxfh.indir` and `rxfh.key` both be 0 here? So this would end up writing to null.

omg, you're right. Thanks for noticing that.

> I'd think it would be fine just setting `.indir = indir` and `.key = key` in the initialiser above.

`indir` and `key` are passed as const, so compiler doesn't allow to do that: `error: initialization discards ‘const’ qualifier from pointer target type`.
Probably we can just discard const by doing like this `.indir = (u32*)indir`. I will recheck it.

Reply:
I've looked through some kernel and sfc netdriver code. It seems we can cast these pointers to non-const safely.
For instance, netdriver developers do the same in their compat code: https://github.com/Xilinx-CNS/onload/blob/6be7c86d813082d4b1bb48961f79e9123e9b572f/src/driver/linux_net/drivers/net/ethernet/sfc/ethtool_common.c#L2105